### PR TITLE
Cross platform and spruce up

### PR DESCRIPTION
**Pull request by slang25**
There's a lot here, feel free to push your own changes.

I've had to keep `FormatSolution` targeting full framework as `MSBuildWorkspace` isn't available on .NET Core yet, and Buildalyzer doesn't seem to give me enough of a functioning workspace to get it working.

### CODE REVIEW

File: bin2hex/bin2hex.csproj [hunk 1]
```diff
@@ -1,43 +1,8 @@
...
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
-  <Import Project="..\Common.props" />
+<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{BD195B29-1385-4AD9-9DEA-3FE0AA237DD2}</ProjectGuid>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <RuntimeIdentifiers>win10-x64;osx.10.11-x64</RuntimeIdentifiers>
```
Comment (KirillOsenkov) [inline bin2hex/bin2hex.csproj:4]: ><RuntimeIdentifiers>win10-x64;osx.10.11-x64</RuntimeIdentifiers> [](start = 4, length = 64)

Curious what is this for? Haven't seen this before.

File: bin2hex/bin2hex.csproj [hunk 2]
```diff
@@ -1,43 +1,8 @@
...
     <OutputType>Exe</OutputType>
-    <AssemblyName>bin2hex</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <VersionPrefix>2.0.0</VersionPrefix>
```
Comment (KirillOsenkov) [inline bin2hex/bin2hex.csproj:6]: ><VersionPrefix> [](start = 4, length = 15)

What's this for and where is it used?

**Comment by KirillOsenkov** [comment]
Thanks!

File: bin2hex/bin2hex.csproj (hunk 1 above)
Comment (slang25) [reply bin2hex/bin2hex.csproj:4]: This is for if you wanted to build [self-contained deployment](https://docs.microsoft.com/en-us/dotnet/core/deploying/#self-contained-deployments-scd)

**Pull request merged**